### ISSUE TITLE
Questions about different algorithms

### ISSUE THREAD

**Issue by yingchengyang** [mazpie/mastering-urlb#3]
Thanks a lot for such wonderful work and codes! I have some questions about the py files used in pretrain and finetune for different algorithms. In my opinion, it follows the bellow table:

| Alg | Pretrain | Finetune |
|---|---|---|
| DreamerV2  | dreamer.py | dreamer.py (the pretrain and finetune codes are the same) |
| ICM | icm.py | dreamer.py |
| Plan2Explore | plan2explore.py | dreamer.py |
| RND | rnd.py | dreamer.py |
| LBS | lbs.py | dreamer.py |
| APT | apt_dreamer.py | dreamer.py |
| DIAYN | diayn_dreamer.py | diayn_dreamer.py |
| APS | aps.py | aps.py |

Is it right? Thanks a lot again.

**Comment by yingchengyang** [comment]
Moreover, I'd like to ask what are the differences between DIAYN and Dreamer during finetune. Will they utilize the learned skills of the pretrain stage during the finetune stage? Thanks again!

**Comment by mazpie** [comment]
Hi,
the table you posted is correct: both APS and DIAYN require using their "custom agent", as they will use the skills during fine-tuning, while the other methods just use a standard Dreamer agent with the pre-trained behavior modules (i.e. actor and/or critic).

You can see that for both DIAYN and APS the skill behavior is decided through the `meta` inputs, which contain the information about the skill to use. This is randomly sampled during the initial steps of fine-tuning, until the `regress_meta` method is called.

**Comment by yingchengyang** [comment]
Thanks a lot! During the finetune stage, APS and DIAYN will learn to choose the skill with the highest return. Is it right? Moreover, how to get the "ours" results (93.59) in Table 1 of the paper?

Thanks again!

**Comment by yingchengyang** [comment]
Also, I'm curious about whether 100k steps in finetune consider action repeat. In other words, assume that the lengths of episodes are 1000 and action_repeat is 2, 100k steps mean 100 episodes or 200 episodes?  In my opinion, it will consider action repeat, i.e., 100k steps mean 100 episodes, is it right?

Thanks a lot!

**Comment by mazpie** [comment]
> During the finetune stage, APS and DIAYN will learn to choose the skill with the highest return. Is it right?

Given the limited number of interactions for fine-tuning, the skill is selected after the seed steps and fine-tuned for the rest of the fine-tuning process. This follows the approach used in the original URLB work.

> In other words, assume that the lengths of episodes are 1000 and action_repeat is 2, 100k steps mean 100 episodes or 200 episodes?

For the FT stage we have 100k steps = 100 episodes of length 1000 frames
Given the action repeat of 2, the number of observations, i.e. the number of time you would call `step` on the environment, per episode is 500

Hope this answers your questions!

**Comment by yingchengyang** [comment]
Thanks a lot! Finally, I'm interested about how to get the optimal result in Table 1 (93.59) of the paper via algorithm Dyna-MPC. Which algorithm is used to combine with MPC?

**Comment by mazpie** [comment]
The results in Table 1 use the LBS exploration strategy.
You can find detailed results in the Appendix

**Comment by yingchengyang** [comment]
Thanks a lot!
